src/mscorlib/Resources/Strings.resx

```diff
@@ -2539,6 +2539,9 @@
   <data name="InvalidOperation_CollectionCorrupted" xml:space="preserve">
     <value>A prior operation on this collection was interrupted by an exception. Collection's state is no longer trusted.</value>
   </data>
+  <data name="InvalidOperation_ConcurrentOperationsNotSupported" xml:space="preserve">
+    <value>Concurrent operations are not supported on non-concurrent collections. A concurrent operation was performed on this collection and corrupted its state. Collection's state is no longer correct.</value>
```

Comment:
Nit: "Collection's state" => "The collection's state"

Reply:
Concurrent reads are supported right? Maybe "Operations that change this collection must have exclusive access" or something like that?

Reply:
Now says:

> Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.

How's that?

Reply:
Sounds fine.